packages/core/src/StateNode.ts

```diff
@@ -465,16 +471,22 @@ class StateNode<
    * @param context Custom context (will override predefined context)
    */
   public withConfig(
-    options: Partial<MachineOptions<TContext, TEvent>>,
+    options: MaybeTypegenMachineOptions<
+      TContext,
+      TEvent,
+      BaseActionObject,
+      TResolvedTypesMeta,
+      true
```

Comment:
At the moment `withConfig` will require all missing implementation to be provided - just like `interpret`, `useMachine`, and `useInterpret`.

This doesn't quite reflect how one can use it - as people can use `withConfig` to "inject" implementations partially (through multiple `withConfig` calls). Supporting this is possible and I've prototypes it [here](https://www.typescriptlang.org/play?#code/C4TwDgpgBAkgzgOQgNwgJwDwBUB8UC8UA2lgLpQQAewEAdgCZzG0rrkD8UAjFAFxQAGANwAoUJCgA1AIYAbAK4Q42PISxEA1hBAB7AGZQyo8dACy6AOYQMpgDRQEqqAHkAtgEtgN+1t0HHUABkDsbg0AAKaDpg2PYA0k5xFNR0jFC++oZQnOpx5PwsqGiiYmGwtHro5sDS2Mk0DEym0gDGABbuLBjStCA4Tlj1qU2tHV2dlWhQpnicpnxQheglnTRoeq3QWGHV0lAA3iJQUNLI0u6y0gBGshAwrmC3rnQ1wO46tHD8h8fHrW8fL5QODANCdCxHKAAX0hHjgcHB90eEGetFe70+30hfxaAMxwNB4MhMJhpQkAFFPvI0BAAIJnC7XW603EY5S4AiGEQiVboDYtMyjToQZxgPHs3ZDRqGHYQGr2ADKEFkBkI+yheB+J1ZgPYWN+xCSnXS2kykWiGCVKvsAHJ-mybThSPwklK0lhdkQAESnc6XG53B5PF7ScVe0je+2A8PZKDmmJWvS2qOfR1EPILJZoYkiUkmONRZDueiB5Go9GAjCQj1yvZUBru2XyyHmosl+i2SGmdzwxFBlEh8WchAfBDyWT+27YT1euEI2gWJHBtGhtnhnAiJzmNBWKvHGvN45a45zvtlwds-UG7Xi-jkygtBQlvfX47xmw9+eL-vl1eA+w+jqnxejgnavscGQGO+rbFhAHZQIBYYbq+OCiAaUJodCm4rGifKbNMQpdAedYpNK2yQLsmqQgA7p4bQAMIfHo7gWHU9bDAR7TCqK4rTrW9i4DgAAU2JQC0TEsfwWCQgAlPwzRcV0MElkuA4rrxxECf0oikry6z4QpYwQIxFQsXxNRUccJhwHqhi7Dp3J6fyZi1gcsKfmey4Vp8tnHjel4IXoOg6F6UAAD4IVc0hoF6mEYbm3Ilo+0XQHo8i0EBYk0qGECGcK5kkQ2TDkbltbCZC4mmRY8mEcZEmscRoGQtE4q2XlLA8WyBX2OqG5yZxRkFah3L5rSNIKjozyqb+4rdr2C4vsRbrFU20hgdMHkLtNF6AsOo7jpO1jEd6p5bT+O3AaQG5OPASBFC+MgKEoGB+emUDGpBG3zd+57qWyzpfV+21-YC6akHF13LdwkKcAA9LD70GOJrhXMK9BQOlGJQJkciyFAp0WO950g7QUDWe9TBZmTbR0BQRQgMAYyE200hMFcEC02AhawfBIDSCAACEokCJC-BcCUSWXDSUBpRleIY3AuW1QVUMlZRInHAAdNrRCuLV9gtf9-BjRAE1TcT3lwHNX5DVDXCiZwuu1TVilHZRBtigF7Uip7laaQcGqkKJ-BO67LuDY1HuteH3G+583UB06Ij9cgOjFiU+b3tI-aSmqkK+oyAbA5bV44regXBaFEVelFMVxe533F3++J+SmQJekFIXhZF0WxcSOklJVIL45yLTZTQ3tCVq1nfFCJxMFnOf8bmMklPDUAAALAHAAC0VCQLie9oFE2byIrk+uKv3JD8A+MAOp0QAYsFnKuJrtGMyZzEWFP+dAUCflO78CEjJAgmooT2F+MSTsUIr4iHXlvXe+8ICH3QCfEQZ8lauyEq4B+jNn46DgTfe+T9gq0gYAAIWiq-PBbQCHvzol-Fiv8y4BT8rXYBoD8DgPWhAlea8EauHkO0KA1EdDUXsMjMAOhFbox6OjTwoidBoA0HAIWmCL60IIeQ+gVC0BwNwaQnQDDP71RYf5QEpcoAcKgCAsBAd1rHAQdvPelAD7ACPugg0ABHai6BQCcPseqXhMC4GZ0oNnZEuwABMnItQF0Ok3curd-4FFYNmY4MITybR+l5ZuADRJtzSUUOKA9r6Alvq4WJhAx4QBypPaeYQAFz1ZlAReUTazRNCQIxYOh8Y5KJr9WsQ44B9MZtARWlU5E7nkOWCmvTb40m8fIdwNJ6AYPPrVHB0Sr5AA) but it comes with a drawback of the auto-completions not working properly at the TS level (in theory, we could bring them back through VScode extension). 

Given this drawback and increased complexity, I've ditched implementing this in XState for the time being. Let's see if people even need it in the first place.

Reply:
The way I implemented it on my initial branch was to make it a deep partial of the options. I think that's a good compromise.

Reply:
But then the user can easily miss the need to provide all implementations, don't they? 🤔

Reply:
Yes, but it's not recommended in most cases to use `.withConfig`, unless for unit testing.